iceoryx2-ffi/cxx/src/node_details.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#include "iox2/node_details.hpp"
+
+namespace iox2 {
+NodeDetails::NodeDetails(NodeName name, const Config& config)
+    : m_node_name { std::move(name) }
```

Comment:
I know it looks weird but it seems to be the new best practice, see: https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html